specification/cognitiveservices/data-plane/AzureOpenAI/authoring/stable/2022-12-01/azureopenai.json

```diff
@@ -0,0 +1,1915 @@
+        "name": "ErrorCode",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "conflict",
+            "description": "The requested operation conflicts with the current resource state."
+          },
+          {
+            "value": "invalidPayload",
+            "description": "The request data is invalid for this operation."
+          },
+          {
+            "value": "forbidden",
+            "description": "The operation is forbidden for the current user/api key."
+          },
+          {
+            "value": "notFound",
+            "description": "The resource is not found."
+          },
+          {
+            "value": "unexpectedEntityState",
+            "description": "The operation cannot be executed in the current resource's state."
+          },
+          {
+            "value": "itemDoesAlreadyExist",
+            "description": "The item does already exist."
+          },
+          {
+            "value": "serviceUnavailable",
+            "description": "The service is currently not available."
+          },
+          {
+            "value": "internalFailure",
+            "description": "Internal error. Please retry."
+          },
+          {
+            "value": "quotaExceeded",
+            "description": "Quota exceeded."
+          },
+          {
+            "value": "jsonlValidationFailed",
+            "description": "Validation of jsonl data failed."
+          },
+          {
+            "value": "fileImportFailed",
+            "description": "Import of file failed."
+          }
+        ]
+      }
+    },
+    "ErrorResponse": {
+      "title": "ErrorResponse",
+      "description": "Error response as defined in the Microsoft REST guidelines\r\n(https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#7102-error-condition-responses).",
+      "type": "object",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/Error"
+        }
+      }
+    },
+    "Event": {
+      "title": "Event",
+      "type": "object",
+      "properties": {
+        "object": {
+          "$ref": "#/definitions/TypeDiscriminator"
+        },
+        "created_at": {
+          "format": "unixtime",
+          "description": "A timestamp when this event was created (in unix epochs).",
+          "type": "integer",
+          "readOnly": true
+        },
+        "level": {
+          "$ref": "#/definitions/LogLevel"
+        },
+        "message": {
+          "description": "The message describing the event. This can be a change of state, e.g., enqueued, started, failed or completed, or other events like uploaded results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "EventList": {
+      "title": "EventList",
+      "description": "Represents a list of events.",
+      "type": "object",
+      "properties": {
+        "object": {
+          "$ref": "#/definitions/TypeDiscriminator"
+        },
```

Comment:
I just noticed:
1. Using "object" as the name of a discriminator is very odd. Normally we use "kind" or "type", and "object" seems to imply that the value is an object which it is not. But if that's the convention used in OpenAI I guess we are stuck with it.
2. I know that events are polymorphic -- they have an "object" discriminator. So why is there also a discriminator in the list? Are all events in the list guaranteed to be of this one type? This is also a very unusual pattern.

Reply:
inherited from OpenAI API